composables/massmint/massMintHelpers.ts

```diff
@@ -0,0 +1,212 @@
+  return createdNFTs
+    .map((nft: CreatedNFT) => {
+      const matchingToken = tokens.find(
+        (token) =>
+          token.name === nft?.name &&
+          token.selectedCollection?.id === nft.collection
+      )
+
+      if (matchingToken === undefined) {
+        $consola.error('matching token not found', nft)
+        return undefined
+      }
+      if (matchingToken.price === undefined) {
+        return undefined
+      }
+      const nftId = getNftId(
+        {
+          ...nft,
+          name: nft.name.toUpperCase().replace(/ /g, '_'),
+          collection: matchingToken.selectedCollection as EntityWithId,
+        },
+        blockNumber
+      )
+      if (nftId === undefined) {
+        return undefined
+      }
+      return {
+        nftId,
+        price: matchingToken.price,
+      }
+    })
+    .filter(Boolean) as TokenToList[]
+}
+
+export const mintKusama = (tokens) => {
+  const { blockNumber, transaction, isLoading, status, isError } =
+    useTransaction()
+  const { urlPrefix } = usePrefix()
+  const createdNFTs = ref<CreatedNFT[]>()
+  transaction({
+    interaction: Interaction.MINTNFT,
+    urlPrefix: urlPrefix.value,
+    token: tokens,
+  }).then((result) => {
+    if (typeof result === 'object' && result?.createdNFTs) {
+      createdNFTs.value = result?.createdNFTs.value as CreatedNFT[]
+    }
+  })
+
+  return {
+    blockNumber,
+    isLoading,
+    status,
+    createdNFTs,
+    isError,
+  }
+}
+
+export const kusamaMintAndList = (tokens) => {
+  const status = ref('')
+  const isLoading = ref(true)
+  const collectionUpdated = ref(false)
+  const blockNumber = ref<string>()
+  const isError = ref(false)
+
+  const {
+    blockNumber: mintBlockNumber,
+    createdNFTs,
+    status: mintStatus,
+    isError: mintIsError,
+  } = mintKusama(tokens)
+  watch([mintBlockNumber, createdNFTs, mintStatus, mintIsError], () => {
+    status.value = mintStatus.value
+    blockNumber.value = mintBlockNumber.value
+    isError.value = mintIsError.value
+
+    if (mintIsError.value) {
+      isLoading.value = false
+    }
+    if (mintBlockNumber.value && createdNFTs.value) {
+      const listForSellItems = getListForSellItems(
+        createdNFTs.value,
+        tokens,
+        mintBlockNumber.value
+      )
+      const {
+        blockNumber: listBlockNumber,
+        isLoading: listIsLoading,
+        status: listStatus,
+        isError: listIsError,
+      } = listForSell(listForSellItems)
+
+      watchEffect(() => {
+        status.value = listStatus.value
+        blockNumber.value = listBlockNumber.value
+        isError.value = listIsError.value
+        if (!listIsLoading.value || listIsError.value) {
+          isLoading.value = false
+        }
+        if (listBlockNumber.value !== undefined) {
+          collectionUpdated.value = true
+        }
+      })
+    }
+  })
+
+  return {
+    status,
+    isLoading,
+    collectionUpdated,
+    blockNumber,
+    isError,
+  }
```

Comment:
>we must have it? I thought we had a callback or something like this on error

https://github.com/kodadot/nft-gallery/pull/5882#discussion_r1210222591

@roiLeo I'm using it here to chain minting and listing on Kusama while giving use a "continues process" feeling

i.e keep the loder spinning for both operation as if they were one